include/cuda/std/detail/libcxx/include/span

```diff
@@ -278,277 +279,232 @@ public:
     constexpr auto subspan() const noexcept
         -> span<element_type, _Count != dynamic_extent ? _Count : _Extent - _Offset>
     {
-        static_assert(_Offset <= _Extent, "Offset out of range in span::subspan()");
-        return {data() + _Offset, _Count == dynamic_extent ? size() - _Offset : _Count};
+        static_assert(_Offset <= _Extent, "span<T, N>::subspan<Offset, Count>(): Offset out of range");
+        static_assert(_Count == dynamic_extent || _Count <= _Extent - _Offset, "span<T, N>::subspan<Offset, Count>(): Offset + Count out of range");
+
+        using _ReturnType = span<element_type, _Count != dynamic_extent ? _Count : _Extent - _Offset>;
```

Comment:
We are now repeating ourselves - can we either roll back this part of the change, or remove the trailing return type from the declaration?

Reply:
That is coming from the libc++ implementation. I would rather stay as close as possible to that implementation if possible.

Shout if you feel strongly about that change

Reply:
ack